api-list.go

```diff
@@ -421,20 +422,23 @@ func (c *Client) listObjectVersions(ctx context.Context, bucketName string, opts
 		var (
 			keyMarker       = ""
 			versionIDMarker = ""
+			preName         = ""
+			preKey          = ""
+			perVersions     []Version
 		)
-
-		for {
-			// Get list of objects a maximum of 1000 per request.
-			result, err := c.listObjectVersionsQuery(ctx, bucketName, opts, keyMarker, versionIDMarker, delimiter)
-			if err != nil {
-				sendObjectInfo(ObjectInfo{
-					Err: err,
+		send := func(vers []Version) {
+			if opts.ReverseVersions {
+				sort.Slice(vers, func(i, j int) bool {
```

Comment:
Instead of re-sorting, just reverse the order. There are internal tie-breakers for versions with the same timestamp, and we should respect that.  Just be sure you only reverse once :)

You can use [slices.Reverse](https://pkg.go.dev/slices#Reverse).